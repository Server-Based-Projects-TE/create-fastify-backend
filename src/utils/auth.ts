import { RouteShorthandOptions } from 'fastify';
import { createError } from './errors';

export const requireUser: RouteShorthandOptions = {
  preValidation: async (request) => {
    console.dir(request.user);
    if (!request.isAuthenticated()) {
      throw createError(401);
    }
  },
};
