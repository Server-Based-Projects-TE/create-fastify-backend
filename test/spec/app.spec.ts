import { buildTestFastify } from 'test/utils';

describe('app', () => {
  const fastify = buildTestFastify();
  afterAll(() => {
    fastify.close();
  });
  it('should receive a 200 on GET /', async () => {
    const response = await fastify.inject({
      method: 'GET',
      url: '/',
    });
    expect(response.statusCode).toEqual(200);
    expect(response.json()).toMatchSnapshot();
  });
  it('should receive a 200 on GET /version', async () => {
    const response = await fastify.inject({
      method: 'GET',
      url: '/version',
    });
    expect(response.statusCode).toEqual(200);
    expect(response.json()).toMatchSnapshot();
  });
});
