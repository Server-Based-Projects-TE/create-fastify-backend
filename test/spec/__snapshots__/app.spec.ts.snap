// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app should receive a 200 on GET / 1`] = `
Object {
  "hello": "world",
}
`;

exports[`app should receive a 200 on GET /version 1`] = `
Object {
  "build": Object {
    "date": "1970-01-01T00:00:00.000Z",
    "number": 0,
  },
  "commit": Object {
    "date": "1970-01-01T00:00:00.000Z",
    "hash": "0000000",
  },
  "name": "create-fastify-backend",
  "version": "0.0.0",
}
`;
